lib/carrierwave/sanitized_file.rb

```diff
@@ -329,6 +328,22 @@ def sanitize(name)
       return name.mb_chars.to_s
     end
 
+    def existing_content_type
+      if @file.respond_to?(:content_type) && @file.content_type
+        @file.content_type.to_s.chomp
+      end
+    end
+
+    def mime_magic_content_type
+      MimeMagic.by_magic(File.open(path)).try(:type) if path
```

Comment:
Feel free to do it. We still don't have a proper solution. Would appreciate it very much.